metadata-ingestion/src/datahub/ingestion/source/aws/dynamodb.py

```diff
@@ -155,6 +161,10 @@ def populate_attribute_definitions(self, table) -> Dict[str, str]:
         attribute_definitions = {
             x["AttributeName"]: x["AttributeType"] for x in raw_attribute_definitions
         }
+        snapshot_attribute_definitions = self.populate_attribute_definitions_from_snapshot()
+        if snapshot_attribute_definitions:
+            attribute_definitions.update(snapshot_attribute_definitions)
+
         if table.get("StreamSpecification", {}).get("StreamEnabled", False):
             stream_arn = table["LatestStreamArn"]
             attribute_definitions.update(
```

Comment:
Didn't think of this situation before, but can we log a warning if there are some unexpected differences?

Reply:
done in [24411e3](https://github.com/Affirm/datahub/pull/29/commits/24411e3d1a6dc6b3df886d61efbf061f43c75dbc)